variety/VarietyWindow.py

```diff
@@ -1357,7 +1357,7 @@ def apply_filters(self, to_set, refresh_level):
                 ):
                     self.post_filter_filename = to_set
                     target_file = os.path.join(
-                        self.wallpaper_folder, "wallpaper-filter-%s.jpg" % Util.random_hash()
+                        self.wallpaper_folder, "wallpaper-filter-%s.png" % Util.random_hash()
```

Comment:
I'm reluctant to accept this change. JPGs are lossy, yes, but the majority of wallpapers and photos are still served in this format, and converting things to PNG will easily bloat file sizes from ~3MB (for a UHD image) to almost 10x that size. That's a lot of additional I/O if Variety is configured to change the wallpaper often.

Reply:
In my limited testing, at least in the case of the Web safe filter in this pull request, the PNG version of the filtered image does not have a significantly higher file size than the JPG version, and in many cases can have an even smaller file size.  For example, a 4160x3120 JPG image of about 3.9 MiB is converted to a 4.5 MiB filtered PNG under the Web safe filter &mdash; so not even close to a 10&times; blowup.  I suppose this is because the Web safe filter uses the ImageMagick `remap` command so that the PNG is saved as an 8-bit per pixel indexed color format, rather than in a 24-bit per pixel format.

However, I see the point that the PNG format might not make sense for all filters, especially for the existing default filters.  Thus, I suggest a filter-by-filter option for the output image format (which defaults to .JPG), or to use the PNG format only when the unfiltered image file is "small enough".